src/org/omegat/core/data/ExternalTMFactory.java

```diff
@@ -128,38 +127,25 @@ public boolean onEntry(TMXReader2.ParsedTu tu, TMXReader2.ParsedTuv tuvSource,
                         return false;
                     }
 
+                    boolean keepForeign = Preferences.isPreferenceDefault(Preferences.EXT_TMX_KEEP_FOREIGN_MATCH, false);
```

Comment:
This introduces unneeded coupling to `Preferences`. Instead `keepForeign` should be an instance field with a setter like `useSlash` or `doSegmenting`.

Reply:
Fixed.